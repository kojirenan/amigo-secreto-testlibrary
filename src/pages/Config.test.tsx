import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import Config from './Config';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe('Página de configuração', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );

    //teste de snapshot
    expect(container).toMatchSnapshot();
  });
});
