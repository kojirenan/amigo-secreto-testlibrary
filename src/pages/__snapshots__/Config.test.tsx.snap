// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Página de configuração deve ser renderizada corretamente 1`] = `
<div>
  <form
    class="flex flex-col gap-y-2 max-lg:items-center px-2 text-sm lg:text-base lg:flex-row lg:justify-center"
  >
    <input
      class="w-80 h-12 bg-white ps-8 shadow-lg max-lg:rounded-full border border-black lg:rounded-l-full focus:outline-orange-500 hover:border-orange-500"
      placeholder="Insira os nomes dos participantes"
      type="text"
      value=""
    />
    <button
      class="w-40 h-12 bg-stone-300 max-lg:rounded-full shadow-lg border border-black cursor-pointer lg:rounded-r-full hover:brightness-75 duration-150"
      disabled=""
    >
      Adicionar
    </button>
    
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Iniciar brincadeira
    </button>
  </footer>
</div>
`;
